### ISSUE TITLE
machines: Lists do not handle resizing well (especially disks)

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#12843]
After the PF4 list widget porting, our lists look like this in a medium-width browser window:

![Screenshot_2019-09-18 Virtual Machines - Rain](https://user-images.githubusercontent.com/10246/65147474-7fd43b00-da1e-11e9-8770-52d07b1b155b.png)

They're also problematic in mobile too:

|||
|-|-|
|![Screen Shot 2019-09-18 at 14 15 47](https://user-images.githubusercontent.com/10246/65147641-dccff100-da1e-11e9-8298-cf33b9285d30.png)|![Screen Shot 2019-09-18 at 14 15 53](https://user-images.githubusercontent.com/10246/65147660-e35e6880-da1e-11e9-8567-2170ea6a59ea.png)|

**Comment by garrett** [comment]
I think the bug was introduced in https://github.com/cockpit-project/cockpit/pull/12573

**Comment by KKoukiou** [comment]
Suggested design for Disks/(NICs should be similar) https://github.com/cockpit-project/cockpit/issues/12946#issuecomment-540061117

**Comment by garrett** [comment]
Revised design is @ https://github.com/cockpit-project/cockpit/pull/13474#issuecomment-602701673 (including mobile).

Quote:

> Since the list is currently a mess on mobile, I considered what we could do to port to PF4 and also have mobile working.
> 
> 1. Make sure the children do not overflow in mobile (and consider whether or not it should overflow in desktop mode too)
> 2. There's no additional indentation horizontally, as we're already space-constrained as-is. Therefore, we need to rely on a bunch of visual clues. (Spacing, alignment, lighter weight separators, font weight, blue highlight on the side, etc.)
> 3. Allow the tab tertiary nav to overflow and center it to make it obvious that it's navigation (offsetting it from the rest)
> 4. Move "Add" actions to the left and sorting to the right (or after, but in this case it isn't needed), as the sorting directly affects the list.
> 5. Use rich cards for the disk sub-list in this case (in other words not a nested table)
> 
> The mobile mode is based on the width of an iPhone SE. In a majority of cases, phones will be a little wider. I targeted the extreme case to make sure the layout would work.
> 
> While this is a mockup for disks, we should consider a very similar layout for the other tabs.
>
> Note: The mobile part of the mockup is very tall. The viewport would be less tall than this — but a person can scroll the whole page easily on a phone. :wink: 
>
> ![vm-disks](https://user-images.githubusercontent.com/10246/77336987-95595300-6d28-11ea-914c-8d6e34a366c1.png)

**Comment by KKoukiou** [comment]
we are not using sublists anymore.

**Comment by garrett** [comment]
@KKoukiou: True, but the lists still don't handle resizing well, mainly with the disks, and for the same problems, and would need the same or siminlar solution (done with PF4 widgets, of course).

It's _mainly_ the file path being so long.

**Comment by garrett** [comment]
Closing, as @KKoukiou mentioned that there's an issue about the lists already.

**Comment by garrett** [comment]
As mentioned in https://github.com/cockpit-project/cockpit/issues/12843 — this isn't sublist related anymore, but still applies to the layout of the list.

**Comment by KKoukiou** [comment]
@garrett this is an design introduced long time before our details page. I am pretty satisfied with how our disks list are now, see screenshot bellow. Shall we just leave it maybe?
![Screen Shot 2021-02-12 at 22 59 37](https://user-images.githubusercontent.com/14921356/107827003-33616c00-6d86-11eb-88ed-0a1b5fe4cee2.png)

**Comment by garrett** [comment]
This is still a problem, especially at various sizes.

![Screenshot_2021-02-15 Virtual Machines - garrett Rain(1)](https://user-images.githubusercontent.com/10246/107970473-70f60d00-6fb1-11eb-9468-af067752b8cf.png)

![Screenshot_2021-02-15 Virtual Machines - garrett Rain](https://user-images.githubusercontent.com/10246/107970477-718ea380-6fb1-11eb-993b-f24a4f6b8afe.png)

(The early breaking into mobile mode was a work-around as to not aggressively wrap the file path.)

I've been looking into a solution, but I think the proper one, subgrid, is still a good ways off on non-Firefox browsers... and it's something we'd want fixed upstream in PatternFly too.

Meanwhile, we could possibly make a workaround by constricting the max width of the labels or stacking them vertically in some situations. 

I suppose we could just ignore it as this is really less problematic in the new layout than before?

**Comment by KKoukiou** [comment]
@garrett right, it's not optimal for all screen size but also it's not really super broken. I suggest with don't bother about it more.

**Comment by garrett** [comment]
Sure. I can't come up with a good fix without hardcoding or using some CSS that only Firefox currently understands. 

The hardcoding would allow the labels to wrap, but we'd need to set the max width at something like 10em (or 15em or whatever), so it's not a generic fix, but would be specific to where we know approximate label sizes. And even then, some languages may wrap.

The alternate approaches are to use subgrid to allow the labels to be constrained to their actual sizes (and we could set a minimum and/or maximum)... but that only works in Firefox right now.

An alternate solution would have been to use display: contents, but that breaks accessibility due to a bug in Chrome. So it's also not workable.

I tried to come up with a fourth alternate solution (first is hardcoding, second is subgrid, third is display: contents), but it starts to veer further and further away from PatternFly, so it's not really viable.

Oh well. :disappointed: 

This is the same issue we have with description lists (https://github.com/cockpit-project/cockpit/pull/14756). Both would be solved with using tables for the tabular data, but that's not how PatternFly is handling this (I hinted about this above), despite the table data literally being called table (and it's using divs instead). :shrug:
